setup.py

```diff
@@ -78,6 +78,7 @@ def _global_version(version):
         "local_scheme": _local_version,
         "version_scheme": _global_version,
         "write_to": "vyper/version.py",
+        "tag_regex": r"^(?:[\w-]+-)?(?P<version>[vV]?\d+(?:\.\d+){0,2}[^-\+]*)(?:[-\+].*)?$"
```

Comment:
by the way, this is very slick. BUT, i wonder if it will cause any issues with pypi tagging (i.e. it will let some non-PEP440 compliant version slip thru, causing pip to reject the upload).